azure-mgmt-network/src/main/java/com/microsoft/azure/management/network/implementation/NetworkImpl.java

```diff
@@ -137,10 +133,14 @@ public NetworkImpl withAddressSpace(String cidr) {
         return this;
     }
 
+    @Override
+    public SubnetImpl defineSubnet(String name) {
+        SubnetInner inner = new SubnetInner();
+        inner.setName(name);
+        return new SubnetImpl(name, inner, this);
```

Comment:
related to previous comment.. we could add SubnetImpl to List<Subnet> here that ensure this list contains all the subnets when we populate the subnets in the payload in `createResource` method.